padertorch/contrib/examples/vae_mnist/mnist_vae_example.py

```diff
@@ -0,0 +1,216 @@
+"""
+A very simple training example for a Variational Autoencoder(VAE)
+using the mnist database, sacred and the padertorch trainer
+
+Use the following command:
+    python mnist_vae_example.py with logfilepath=/path/to/dir/ epochs=500
+
+Other options for sacred are available, refer to the config() function.
+"""
+import numpy as np
+import os
+
+import padertorch as pt
+import padertorch.train.optimizer as pt_opt
+
+import sacred
+
+import torch
+import torch.nn as nn
+
+from einops import rearrange
+
+from padertorch.data.utils import collate_fn
+from padertorch.ops.losses.kl_divergence import gaussian_kl_divergence
+from padertorch.ops.mappings import ACTIVATION_FN_MAP
+from padertorch.testing.test_db import MnistDatabase
+
+from torch.distributions import Normal
+
+ex = sacred.Experiment()
+
+
+def get_dataset(batch_size):
+    """
+    load MNIST train and test dataset
+
+    Args:
+        batch_size (int):
+
+    Returns:
+        train_set and test_set
+
+    """
+    db = MnistDatabase()
+    train_set = db.get_dataset('train')
+    test_set = db.get_dataset('test')
+
+    training_data = prepare_dataset(train_set, batch_size=batch_size, training=True)
+    test_data = prepare_dataset(test_set, batch_size=batch_size, training=False)
+
+    return training_data, test_data
+
+
+def prepare_dataset(dataset, batch_size, training=False):
+    if training:
+        dataset = dataset.shuffle(reshuffle=True)
+
+    def Collate(batch):
+        batch = collate_fn(batch)
+        batch['image'] = np.array([tensor for tensor in batch['image']])
+        return batch
+
+    return dataset.prefetch(
+        num_workers=1, buffer_size=10*batch_size
+    ).batch(batch_size=batch_size).map(Collate)
+
+
+class VAEModel(pt.Model):
+    def __init__(self, activation_function, hidden_channels, kernel_size):
+        super().__init__()
+        self.net = VAE(activation_function, hidden_channels, kernel_size)
+
+    def forward(self, inputs):
+        images = inputs['image'][:, None, :, :]
+        out, mu, std = self.net(images)
+
+        return dict(
+            input=images,
+            mean=mu,
+            prediction=out,
+            std=std
+        )
+
+    def review(self, inputs, outputs):
+        images = rearrange(outputs['input'], 'b c x y -> b (c x y)')
+        prediction = rearrange(outputs['prediction'], 'b c x y -> b (c x y)')
+
+        mse = nn.functional.mse_loss(prediction, images)  # reconstruction error
+
+        q = Normal(loc=outputs['mean'], scale=outputs['std'])  # posterior distribution
+        mean = torch.zeros(outputs['mean'].shape[-1])
+        std = torch.ones(outputs['std'].shape[-1])
+        p = Normal(loc=mean, scale=std)  # prior distribution zero mean and unit variance
+        kld = gaussian_kl_divergence(q=q, p=p)  # regularisation error
+
+        loss = mse + torch.sum(kld)
+
+        return dict(
+            loss=loss,
+        )
```

Comment:
it would be nice if some images would be added (originals and reconstructions) to the review which can be viewed in the tensorboard.